internal/component/discovery/discovery.go

```diff
@@ -74,16 +76,26 @@ type Component struct {
 	latestDisc    DiscovererWithMetrics
 	newDiscoverer chan struct{}
 
-	creator Creator
+	creator            Creator
+	debugDataPublisher livedebugging.DebugDataPublisher
 }
 
+var _ component.Component = (*Component)(nil)
+var _ component.LiveDebugging = (*Component)(nil)
+
 // New creates a discovery component given arguments and a concrete Discovery implementation function.
 func New(o component.Options, args component.Arguments, creator Creator) (*Component, error) {
+	debugDataPublisher, err := o.GetServiceData(livedebugging.ServiceName)
+	if err != nil {
```

Comment:
If there is an error in connecting with livedebugging I would not expect that to cause the component to fail to instantiate.

Just looked, and this is clearly the established pattern so I'll bring that up in a separate channel, this is fine for now.